lms/djangoapps/commerce/views.py

```diff
@@ -148,9 +149,37 @@ def checkout_cancel(_request):
 @login_required
 def checkout_receipt(request):
     """ Receipt view. """
+
+    page_title = _('Receipt')
+    is_payment_complete = True
+    is_cybersource = all(k in request.POST for k in ('signed_field_names', 'decision', 'reason_code'))
+    if is_cybersource and request.POST['decision'] != 'ACCEPT':
+        # Cybersource may redirect users to this view if it couldn't recover
+        # from an error while capturing payment info.
+        is_payment_complete = False
+        page_title = _('Payment Failed')
+        reason_code = request.POST['reason_code']
```

Comment:
In order to shorten the line below this:
```python
reason_code = int(request.POST['reason_code'])
```
To prevent 500s in case CyberSource does something silly and doesn't pass a `reason_code`:
```python
reason_code = int(request.POST.get('reason_code', 0))
```